decomposer/neuro_san_solver.py

```diff
@@ -56,29 +56,62 @@ def __init__(self, winning_vote_count: int = 2,
         if self.solution_candidate_count is None:
             self.solution_candidate_count = default_count
 
-        session: AgentSession = None
         self.parsing = SolverParsing()
 
-        # Initialize the Neuro SAN agent session callers
-        session = SessionManager.get_session("composition_discriminator")
-        self.composition_discriminator_caller: AgentCaller = NeuroSanAgentCaller(session, self.parsing)
+        self.composition_discriminator_caller: AgentCaller = None
+        self.decomposer_caller: AgentCaller = None
+        self.problem_solver_caller: AgentCaller = None
+        self.solution_discriminator_caller: AgentCaller = None
+
+    def set_callers(self, composition_discriminator_caller: AgentCaller,
+                    decomposer_caller: AgentCaller,
+                    problem_solver_caller: AgentCaller,
+                    solution_discriminator_caller: AgentCaller):
+        """
+        Set AgentCallers.
+        """
+
+        if composition_discriminator_caller is not None:
+            self.composition_discriminator_caller = composition_discriminator_caller
+        if decomposer_caller is not None:
+            self.decomposer_caller = decomposer_caller
+        if problem_solver_caller is not None:
+            self.problem_solver_caller = problem_solver_caller
+        if solution_discriminator_caller is not None:
+            self.solution_discriminator_caller = solution_discriminator_caller
+
+    def set_callers_if_not_already_set(self):
+        """
+        Initialize AgentCallers if they are not already set.
+        """
```

Comment:
Create a method to initialize the AgentCallers as they were before in the constructor.